### ISSUE TITLE
Cypress uses the system Node.js when `nodeVersion` is set to `bundled`

### ISSUE THREAD

**Issue by JessefSpecialisterren** [cypress-io/cypress#27016]
### Current behavior

When the `nodeVersion` configuration option is set to `bundled`, Cypress uses the system Node.js to execute `setupNodeEvents` and tasks

### Desired behavior

I see two ways of fixing this bug:
1. Fix the `nodeVersion` configuration option so that `bundled` works as it used to, and Cypress uses the bundled Node.js to execute `setupNodeEvents` and tasks
2. Remove the `nodeVersion` configuration option altogether, as has been the stated intention since Cypress 9.0.0

My team and I would prefer solution 1, as stated and explained in #18882 . However, that issue didn't gain a lot of traction, so if you choose solution 2 I can understand that

### Test code to reproduce

cypress.config.js:
```js
module.exports = {
  e2e: {
    nodeVersion: 'bundled',
    setupNodeEvents(on, config) {
      console.log(process.version)
      on('task', {
        logNodeVersion() {
          console.log(process.version)
          return true
        }
      })
    },
  },
}
```

spec.cy.js:
```js
it('', () => {
  cy.task('logNodeVersion')
})
```

Make sure you have a system Node.js with a different version than the bundled Node.js. Note the version logged in the terminal when you select E2E testing and when you run spec.cy.js

### Cypress Version

12.14.0

### Node version

v18.16.0

### Operating System

Windows 10.0.19044.2965

**Comment by mschile** [comment]
Hi @JessefSpecialisterren 👋, thanks for taking the time to log this issue. I was able to reproduce your issue and will forward this ticket to the appropriate team. They will evaluate the priority of this ticket and consider their capacity to pick it up. As of right now, the plan is to still remove the `nodeVersion` configuration in a future major release. It looks like this issue was introduced in Cypress v10. Thanks!

**Comment by jennifer-shehane** [comment]
The code for this is done, but this has yet to be released. We'll update this issue and reference the [changelog](https://on.cypress.io/changelog) when it's released.

**Comment by jennifer-shehane** [comment]
Released in [Cypress 13.0.0](https://docs.cypress.io/guides/references/changelog#13-0-0).
